pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -78,106 +66,48 @@ func (r *ReconcileKogitoApp) Reconcile(request reconcile.Request) (reconcile.Res
+		return reconcile.Result{}, err
+	} else if !imageExists {
+		// let's wait for the build to finish
+		return reconcile.Result{RequeueAfter: time.Duration(30) * time.Second}, nil
 	}
 
-	for buildType, buildConfig := range buildConfigs.AsMap {
-		if _, err := r.ensureImageStream(
-			buildConfig.Name,
-			instance,
-		); err != nil {
+	// checks for dc updates
+	if kogitoInv.DeploymentConfig != nil {
+		if err := controllerutil.SetControllerReference(instance, kogitoInv.DeploymentConfig, r.scheme); err != nil {
 			return reconcile.Result{}, err
 		}
-
-		// Check if this BC already exists
-		_, err = r.buildClient.BuildConfigs(buildConfig.Namespace).Get(buildConfig.Name, metav1.GetOptions{})
-		if err != nil && errors.IsNotFound(err) {
-			log.Info("Creating a new BuildConfig ", buildConfig.Name, " in namespace ", buildConfig.Namespace)
-			bc, err := r.buildClient.BuildConfigs(buildConfig.Namespace).Create(buildConfig)
-			if err != nil {
-				return reconcile.Result{}, err
-			}
-
-			// Trigger first build of new builder BC
-			if buildType == defs.S2IBuildType {
-				if err = r.triggerBuild(*bc, instance); err != nil {
-					return reconcile.Result{}, err
-				}
-			}
-		} else if err != nil {
+		if dcUpdated, err := r.updateDeploymentConfigs(instance, *kogitoInv.DeploymentConfig); err != nil {
 			return reconcile.Result{}, err
+		} else if dcUpdated && status.SetProvisioning(instance) {
+			return r.UpdateObj(instance)
 		}
 	}
 
-	// Create new DeploymentConfig object
-	depConfig, err := r.newDCForCR(instance, buildConfigs.AsMap[defs.RunnerBuildType], &sa)
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-	rResult, err = r.createObj(
-		&depConfig,
-		r.client.Get(context.TODO(), types.NamespacedName{Name: depConfig.Name, Namespace: depConfig.Namespace}, &oappsv1.DeploymentConfig{}),
-	)
-	if err != nil {
-		return rResult, err
-	}
-
-	dcUpdated, err := r.updateDeploymentConfigs(instance, depConfig)
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-	if dcUpdated && status.SetProvisioning(instance) {
-		return r.UpdateObj(instance)
-	}
-
 	// Expose DC with service and route
-	service := defs.NewService(instance, &depConfig)
-	if service != nil {
-		err = controllerutil.SetControllerReference(instance, service, r.scheme)
+	if kogitoInv.Service != nil {
+		err = controllerutil.SetControllerReference(instance, kogitoInv.Service, r.scheme)
 		if err != nil {
 			log.Error(err)
 		}
-
-		service.ResourceVersion = ""
-		rResult, err = r.createObj(
-			service,
-			r.client.Get(context.TODO(), types.NamespacedName{Name: service.Name, Namespace: service.Namespace}, &corev1.Service{}),
-		)
-		if err != nil {
-			return rResult, err
-		}
-
-		// Create route
-		rt, err := defs.NewRoute(instance, service)
-		if err != nil {
-			return rResult, err
-		}
-		if serviceRoute := r.GetRouteHost(*rt, instance); serviceRoute != "" {
+		// TODO: this is really needed?
```

Comment:
It's a usability requirement, and not a functional one. The status/route in the CR (instance) is used [to populate](https://github.com/kiegroup/kogito-cloud-operator/blob/master/deploy/catalog_resources/community/0.1.0/kogitocloud-operator.0.1.0.clusterserviceversion.yaml#L83-L87) the OLM page. This can only be done from the content of the CR, and the value can only be obtained dynamically based on the OpenShift cluster settings. The end result is that someone using OLM can easily and directly find the URL of the kogito service that was deployed.